cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/workflow/WorkflowProgramRunner.java

```diff
@@ -140,6 +144,8 @@ public ProgramController run(final Program program, final ProgramOptions options
       // service can be fully captured by the controller.
       final ProgramController controller = new WorkflowProgramController(program, driver, serviceAnnouncer, runId);
       final String twillRunId = options.getArguments().getOption(ProgramOptionConstants.TWILL_RUN_ID);
+      final RetryStrategy retryStrategy = SystemArguments.getRetryStrategy(options.getUserArguments().asMap(),
```

Comment:
what would happen if we retry indefinitely instead of retry according to the program type retry policy? Would it just keep retrying until the program ends? Now that I think of it, this is really more of a system operation that has nothing to do with what type of program is running. For example, it doesn't make sense for mapreduce to retry this for 10 minutes but for services not to retry at all.